backend/controllers/followController.js

```diff
@@ -0,0 +1,195 @@
+const { User, ActivityLog, Follow } = require("../models");
+
+const putfollowAndUnfollow = async (req, res, next) => {
+  const { id } = req.params;
+  const user_id = req.user;
+  try {
+    const check = await Follow.findOne({
+      where: { follower_id: user_id, following_id: id },
+      order: [["createdAt", "DESC"]],
+    });
+
+    if (check && check.flag) {
+      await Follow.update({ flag: false }, { where: { id: check.id } });
+      await ActivityLog.create({
+        relatable_id: check.id,
+        relatable_type: "unfollows",
+        user_id,
+      });
+      res.status(200).json({ message: "Unfollow operation succeded" });
+    } else if (check && check.flag === false) {
+      await Follow.update({ flag: true }, { where: { id: check.id } });
+      await ActivityLog.create({
+        relatable_id: check.id,
+        relatable_type: "follows",
+        user_id,
+      });
+      res.status(200).json({ message: "Follow operation succeded" });
+    } else {
+      const follow = await Follow.create({
+        follower_id: user_id,
+        following_id: id,
+        flag: true,
+      });
+      await ActivityLog.create({
+        relatable_id: follow.id,
+        relatable_type: "follows",
+        user_id: user_id,
+      });
+      res.status(200).json({ message: "Follow operation succeded" });
+    }
+  } catch (err) {
+    res
+      .status(400)
+      .json({ error: "Something went wrong, please try again later" });
+  }
+};
+
+const getFollowsCount = async (req, res, next) => {
+  const { id } = req.params;
+  try {
+    const following = await Follow.findAll({
+      where: { follower_id: id, flag: true },
+    });
+    let followingCount = 0;
+    await Promise.all(
+      following.map(async ({ following_id }) => {
+        const user = await User.findByPk(following_id);
+        if (user) {
+          followingCount++;
+        }
+      })
+    );
+
+    const follower = await Follow.findAll({
+      where: { following_id: id, flag: true },
+    });
+    let followerCount = 0;
+    await Promise.all(
+      follower.map(async ({ follower_id }) => {
+        const user = await User.findByPk(follower_id);
+        if (user) {
+          followerCount++;
+        }
+      })
+    );
+
+    res.status(200).json({
+      followers: followerCount,
+      following: followingCount,
+    });
+  } catch (err) {
+    res
+      .status(400)
+      .json({ error: "Something went wrong, please try again later" });
+  }
+};
+
+const getFollowing = async (req, res, next) => {
+  const { id } = req.params;
+  const user_id = req.user;
+  const followingsList = [];
+
+  try {
+    const followingIdList = await Follow.getFollowingIdLIst(user_id);
+
+    const { Follows } = await User.findByPk(id, {
+      include: [{ model: Follow, attributes: ["following_id", "flag"] }],
+    });
+
+    if (Follows.length > 0) {
```

Comment:
- minimize tabbing as possible by using negate conditioning.
- goal is to avoid blocks/multiple lines being tabbed (tabbing is not bad practice though but it will help other devs to read your code faster)
```
if (Follows.length <= 0) {
  return res.status(200).json({ following: followingsList });
}

// else if greater than zero, execute the statements below.
// In this case we prevented multiple lines of code being tabbed.
```
- apply to all related